src/scripts/main.js

```diff
@@ -1,3 +1,221 @@
+    setSortTable(index);
+  });
+});
+
+const setSortTable = (columnIndex) => {
+  updateRows();
+
+  const sortedRows = rows.sort((a, b) => {
+    const aValue = a.cells[columnIndex].textContent.trim();
+    const bValue = b.cells[columnIndex].textContent.trim();
+
+    if (+aValue) {
+      return sortOrder[columnIndex] ? +aValue - bValue : bValue - aValue;
+    }
+
+    if (columnIndex === 4) {
+      const aSalary = parseFloat(aValue.replace(/[$,]/g, ''));
+      const bSalary = parseFloat(bValue.replace(/[$,]/g, ''));
+
+      return sortOrder[columnIndex] ? aSalary - bSalary : bSalary - aSalary;
+    }
+
+    return sortOrder[columnIndex] ? aValue.localeCompare(bValue) : bValue.localeCompare(aValue);
+  });
+
+  tBody.innerHTML = '';
+  tBody.append(...sortedRows);
+  sortOrder[columnIndex] = !sortOrder[columnIndex];
+};
+
+//Selecting row:
+
+[...tBody.rows].forEach((row) => {
+  row.addEventListener('click', (e) => {
+    const rowsActive = tBody.querySelectorAll('tr.active');
+    rowsActive.forEach(rowActive => {
+      rowActive.classList.remove('active');
+    });
+
+    row.classList.toggle('active');
+  });
+
+})
+
+//Create form: 
+
+const form = document.createElement('form');
+form.className = 'new-employee-form';
+
+form.innerHTML = `
+  <label>
+    Name: <input name="name" type="text" data-qa="name" required>
+  </label>
+  <label>
+    Position: <input name="position" type="text" data-qa="position" required>
+  </label>
+  <label>
+    Office:
+    <select name="office" data-qa="office" required>
+      <option>Tokyo</option>
+      <option>Singapore</option>
+      <option>London</option>
+      <option>New York</option>
+      <option>Edinburgh</option>
+      <option>San Francisco</option>
+    </select>
+  </label>
+  <label>
+    Age: <input name="age" type="number" data-qa="age" required>
+  </label>
+  <label>
+    Salary: <input name="salary" type="number" data-qa="salary" required>
+  </label>
+  <button type="button">Save to table</button>
+`;
+
+body.append(form);
+
+form.querySelector('button').addEventListener('click', (e) => {
+  // e.preventDefault();
+
+  if (!checkFormFields()) {
+    return;
+  }
+  
+  const newRow = tBody.insertRow();
+
+  if (
+    form.elements.name.value &&
+    form.elements.position.value &&
+    form.elements.office.value &&
+    form.elements.age.value &&
+    form.elements.salary.value
+  ) {
+      newRow.insertCell(0).textContent = form.elements.name.value.trim();  
+      newRow.insertCell(1).textContent = form.elements.position.value.trim();
+      newRow.insertCell(2).textContent = form.elements.office.value.trim();
+      newRow.insertCell(3).textContent = parseInt(form.elements.age.value, 10);
+      let salaryFormated = parseFloat(form.elements.salary.value.replace(/[^0-9.]/g, ''));
+      newRow.insertCell(4).textContent = `$${new Intl.NumberFormat("us-IN").format(salaryFormated)}`;
+  }
+
+  checkFormFields();
+  tBody.append(newRow);
```

Comment:
The call to `checkFormFields()` here is redundant because it has already been called before appending the new row. Consider removing this line to avoid unnecessary checks.